crates/burn-jit/src/kernel/reduce/tune.rs

```diff
@@ -0,0 +1,218 @@
+use burn_tensor::ElementConversion;
+use cubecl::{
+    client::ComputeClient,
+    tune,
+    tune::{local_tuner, tune_with, LocalTuner},
+    AutotuneKey,
+};
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    kernel::prng::random_like_uniform, ops::numeric::empty_device, tensor::JitTensor,
+    JitAutotuneKey, JitElement, JitRuntime, JitTuneId,
+};
+
+/// Executes autotune on reduce operations.
+pub fn autotune_reduce<
+    Run: JitRuntime,
+    In: JitElement,
+    Out: JitElement,
+    Rd: cubecl::reduce::Reduce,
+>(
+    client: &ComputeClient<Run::Server, Run::Channel>,
+    input: JitTensor<Run>,
+    output: JitTensor<Run>,
+    dim: usize,
+) -> Result<(), cubecl::reduce::ReduceError> {
+    static TUNER: LocalTuner<JitAutotuneKey, JitTuneId> = local_tuner!();
+
+    TUNER.execute(
+        &JitTuneId::new::<Run>(&input.device),
+        client,
+        Box::new(ReduceOps::<Run, In, Out, Rd>::new(input, output, dim)),
+    );
+
+    Ok(())
+}
+
+#[derive(Hash, Eq, PartialEq, Debug, Clone, Serialize, Deserialize, AutotuneKey)]
+/// Autotune key representative of redue versions
+pub struct ReduceAutotuneKey {
+    dtype: burn_tensor::DType,
+    #[autotune(anchor)]
+    reduce_axis_shape: usize,
+    #[autotune(anchor)]
+    reduce_axis_stride: usize,
+    #[autotune(anchor)]
+    outer_axes_product: usize, // The product of the shapes of all axes with greater strides.
+}
+
+impl ReduceAutotuneKey {
+    pub(crate) fn generate<Run: JitRuntime>(input: &JitTensor<Run>, axis: usize) -> Self {
+        let rank = input.shape.num_dims();
+
+        if axis > rank {
+            panic!("axis {axis} is out-of-bound for a rank of {rank}");
+        }
+
+        let dtype = input.dtype;
+        let reduce_axis_shape = input.shape.dims[axis];
+        let reduce_axis_stride = input.strides[axis];
+
+        let outer_axes_product = input
+            .strides
+            .iter()
+            .zip(input.shape.dims.iter())
+            .filter_map(|(stride, shape)| (*stride > reduce_axis_stride).then_some(shape))
+            .product();
+
+        Self {
+            dtype,
+            reduce_axis_shape,
+            reduce_axis_stride,
+            outer_axes_product,
+        }
+    }
+}
+
+pub(crate) fn create_key<Run: JitRuntime>(
+    input: &JitTensor<Run>,
+    _output: &JitTensor<Run>,
+    dim: &usize,
+) -> JitAutotuneKey {
+    JitAutotuneKey::Reduce(ReduceAutotuneKey::generate(input, *dim))
+}
+
+#[tune(
+    operations(reduce_shared_false_plane_false, reduce_shared_true_plane_false, reduce_shared_false_plane_true, reduce_shared_true_plane_true),
```

Comment:
Instead of `shared_bool` and `plane_bool` I would add those qualifier only when true:

```
operations(reduce, reduce_shared, reduce_plane, reduce_shared_plane),
```

Reply:
I can't use reduce as the identifier is already used.